### PR TITLE
feat(Toggle)!: refactor to emotion and add focus styles

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Small refactor on Toggle. 
- Uses a pseudo element for the focus state (this is favorable over other options since the element uses solid background colors and does not have a border to replace.
- Cleans up the styles a bit and renames props to be able to be extended with `styled(Toggle)` by removing `theme` prop.

* Refactors `Toggle` to emotion and adds focus states.
* Adds more stories for different states.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: WEB-1077 + A11Y-433
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.8-alpha.2b81e5.0<br/>@codecademy/gamut-labs@7.10.9-alpha.2b81e5.0<br/>@codecademy/gamut@19.11.4-alpha.2b81e5.0<br/>@codecademy/styleguide@21.15.8-alpha.2b81e5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b2dca5dcdea0edb953e6e--gamut-preview.netlify.app>https://600b2dca5dcdea0edb953e6e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b2dca5dcdea0edb953e6e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.8-alpha.8a8d82.0<br/>@codecademy/gamut-labs@7.11.2-alpha.8a8d82.0<br/>@codecademy/gamut@19.11.4-alpha.8a8d82.0<br/>@codecademy/styleguide@21.16.2-alpha.8a8d82.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b3010e75f9a17f1a88852--gamut-preview.netlify.app>https://600b3010e75f9a17f1a88852--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b3010e75f9a17f1a88852>Deploy Logs</a></p>

File: packages/gamut/src/Toggle/index.tsx [hunk 1]
```diff
@@ -1,18 +1,91 @@
...
+import { screenReaderOnly } from '@codecademy/gamut-styles';
+import styled from '@emotion/styled';
 import React, { Component } from 'react';
 
-import styles from './styles/index.module.scss';
+import { Box } from '../Box';
+import { HiddenText } from '../HiddenText';
+
+export type ToggleSizes = keyof typeof sizes;
+export type ToggleVariants = keyof typeof colors;
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Toggle/index.tsx:9]: 👌 👌 👌

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.8-alpha.89a07e.0<br/>@codecademy/gamut-labs@7.11.2-alpha.89a07e.0<br/>@codecademy/gamut@19.11.4-alpha.89a07e.0<br/>@codecademy/styleguide@21.16.2-alpha.89a07e.0<br/></pre>

File: packages/styleguide/.storybook/addons/system/enhancers.ts [hunk 2]
```diff
@@ -36,7 +36,10 @@ const formatSystemProps: ArgTypesEnhancer = ({ parameters }) => {
   const { argTypes } = parameters;
   return mapValues((arg) => {
     // Update theme props
-    if (arg.name === 'theme' && arg.table.type.summary.indexOf('Theme') > -1) {
+    if (
+      arg.name === 'theme' &&
+      arg?.table?.type?.summary.indexOf('Theme') > -1
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/.storybook/addons/system/enhancers.ts:41]: 👍

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM! Should this be named as a breaking change?

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b333de75f9a1b42a88cf9--gamut-preview.netlify.app>https://600b333de75f9a1b42a88cf9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b333de75f9a1b42a88cf9>Deploy Logs</a></p>

**Comment by codecaaron** [comment]
> LGTM! Should this be named as a breaking change?

Oh damn, very right!  Changing.  Nice catch :D

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b3c35df37f027227576ef--gamut-preview.netlify.app>https://600b3c35df37f027227576ef--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b3c35df37f027227576ef>Deploy Logs</a></p>

**Comment by jcenglish** [comment]
Does the toggle change color in different states? In storybook, it's the same color for true and false, only changing to gray when disabled.

**Comment by codecaaron** [comment]
> Does the toggle change color in different states? In storybook, it's the same color for true and false, only changing to gray when disabled.

You are correct.  I think the SB link is stale.  I'm republishing <3

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b52e011748a3cd975379c--gamut-preview.netlify.app>https://600b52e011748a3cd975379c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b52e011748a3cd975379c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.8-alpha.33b35a.0<br/>@codecademy/gamut-labs@7.13.1-alpha.33b35a.0<br/>@codecademy/gamut@19.11.4-alpha.33b35a.0<br/>@codecademy/styleguide@21.18.1-alpha.33b35a.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.8-alpha.a9deb5.0<br/>@codecademy/gamut-labs@7.13.2-alpha.a9deb5.0<br/>@codecademy/gamut@19.11.4-alpha.a9deb5.0<br/>@codecademy/styleguide@21.18.2-alpha.a9deb5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600b58fd5dcdea53a69536d2--gamut-preview.netlify.app>https://600b58fd5dcdea53a69536d2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600b58fd5dcdea53a69536d2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.8-alpha.071aac.0<br/>@codecademy/gamut-labs@7.13.2-alpha.071aac.0<br/>@codecademy/gamut@19.11.4-alpha.071aac.0<br/>@codecademy/styleguide@21.18.2-alpha.071aac.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600c927e1f0dc70cce9aa2c9--gamut-preview.netlify.app>https://600c927e1f0dc70cce9aa2c9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600c927e1f0dc70cce9aa2c9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.8-alpha.c3a0b7.0<br/>@codecademy/gamut-labs@7.13.2-alpha.c3a0b7.0<br/>@codecademy/gamut@19.11.4-alpha.c3a0b7.0<br/>@codecademy/styleguide@21.19.1-alpha.c3a0b7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600dc2450b6423e401dc9dc2--gamut-preview.netlify.app>https://600dc2450b6423e401dc9dc2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600dc2450b6423e401dc9dc2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.9-alpha.ee33a0.0<br/>@codecademy/gamut-labs@7.13.3-alpha.ee33a0.0<br/>@codecademy/gamut@19.11.5-alpha.ee33a0.0<br/>@codecademy/styleguide@21.20.1-alpha.ee33a0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600f15fb7116440c0daa463d--gamut-preview.netlify.app>https://600f15fb7116440c0daa463d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600f15fb7116440c0daa463d>Deploy Logs</a></p>

**Comment by jcenglish** [comment]
One last thing... should the disabled toggle have a disabled cursor (`not-allowed`)?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.13-alpha.9b614e.0<br/>@codecademy/gamut-labs@7.16.1-alpha.9b614e.0<br/>@codecademy/gamut@19.12.1-alpha.9b614e.0<br/>@codecademy/styleguide@21.24.3-alpha.9b614e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601307cf6ddb7156fa59acb2--gamut-preview.netlify.app>https://601307cf6ddb7156fa59acb2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601307cf6ddb7156fa59acb2>Deploy Logs</a></p>

**Pull request merged**